### PR TITLE
Add service_account to google_cloudbuild_trigger

### PR DESCRIPTION
**Pull request by pasha-gh**
Fix [hashicorp/terraform-provider-google#10053](https://github.com/hashicorp/terraform-provider-google/issues/10053).

Add service_account to google_cloudbuild_trigger (Included in the [REST documentation](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.triggers)).

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudbuild: added field `service_account` to `google_cloudbuild_trigger`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Looks good at first glance- /gcbrun to test.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  service_account = "projects/my-project/serviceAccounts/my-service-account"
+  filename        = "cloudbuild.yaml"
+}
+
+resource "google_service_account" "cloudbuild_service_account" {
+  account_id = "my-service-account"
+}
+
+resource "google_project_iam_member" "act_as" {
+  project = google_project_service.agent_project.project
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:16]:
```suggestion
```

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  role    = "roles/iam.serviceAccountUser"
+  member  = "serviceAccount:${google_service_account.cloudbuild_service_account.email}"
+}
+
+resource "google_project_iam_member" "logs_writer" {
+  project = google_project_service.agent_project.project
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:22]:
```suggestion
```

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+resource "google_cloudbuild_trigger" "<%= ctx[:primary_resource_id] %>" {
+  trigger_template {
+    branch_name = "master"
+    repo_name   = "my-repo"
+  }
+
+  service_account = "projects/my-project/serviceAccounts/my-service-account"
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:7]: You'll want to reference the service account here, similar to below- the `id` field should hold the right format, I think.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  filename        = "cloudbuild.yaml"
+}
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:9]: I expect you'll want this resource to get provisioned only once the service account has the right permissions, too. There's no "natural" value to reference to establish a dependency-- instead, let's use https://www.terraform.io/docs/language/meta-arguments/depends_on.html to reference both `*_iam_member` resources

**Review by rileykarson — COMMENTED** [review]
As discussed in chat- neither of these project references are needed, so they can be removed.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb (hunk 1 above)
Comment (pasha-gh) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb]: Done.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb (hunk 2 above)
Comment (pasha-gh) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb]: Done.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb (hunk 3 above)
Comment (pasha-gh) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb]: Done. Changed to `"${google_service_account.cloudbuild_service_account.id}"`.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb (hunk 4 above)
Comment (pasha-gh) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:13]: Done.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+resource "google_cloudbuild_trigger" "<%= ctx[:primary_resource_id] %>" {
+  trigger_template {
+    branch_name = "master"
+    repo_name   = "my-repo"
+  }
+
+  service_account = "${google_service_account.cloudbuild_service_account.id}"
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb:7]:
```suggestion
  service_account = google_service_account.cloudbuild_service_account.id
```

Mostly a nitpick- your syntax is required to interpolate inn the middle of a string, but there's a nicer one that's recommended in `0.12+`.

Comment (pasha-gh) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_service_account.tf.erb]: Updated.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**